### ISSUE TITLE
Refactor for testability

### ISSUE THREAD

**Issue by kaj** [KTH/lms-functions#18]
Here's an idea to improve testability, and probably also make it more efficient and easier on canvas.

The program room PR #16 refactors main to look like:

```typescript
  did_a = enrollSomeIfApplicable(message);
  did_b = enrollOtherIfApplicable(message);
  did_c = enrollThirdIfApplicable(message);
  if (!didA && !didB && !didC) {
      context.log("Message is not relevant. Skipping...");
  }
```

Here, all of the `enroll...IfApplicable` methods sends csv file if there is anything to do, so they all need the ability to talk to canvas.

I propose instead:

```typescript
   const enrollments = [];
   enrollments.push(...enrollmentForSomeIfApplicable(message));
   enrollments.push(...enrollmentForOtherIfApplicable(message));
   enrollments.push(...enrollmentForThirdIfApplicable(message));
   
   if (enrollments.length > 0) {
       await canvasApi.endEnrollments(enrollments);
   } else {
       context.log("Message is not relevant. Skipping...");
   }
```

Here, each `enrollmentFor...IfApplicable` is a pure function (except for logging), that takes a message and returns an array with one, two or zero `StudentEnrollment`s, makeing more complete testing easier to do.

Also, with this refactorization, a normal course registration on a program would send one csv file with three data lines instead of two separate files (with 2 + 1 data lines) to Canvas.

**Comment by kaj** [comment]
> This would make it more readable _now_, since all we do in this function _at the moment_ is enrollments. But what if we want to add another feature to this function that does something else then enrollments?

You kind of have a point there, but I'm not convinced.  For one thing, making things clearer now should not be stopped by _possible_ future generalization, only things that we _probably_ will add support for in the nearest year or so.

For another thing, if we _do_ add something entirely different, I think it would be a good idea to "group" the things, so that should be done by extracting the suggested lines from main as `handleEnrollments` and add one specific call `handleTheOtherThing` beside the call to that.  That way, the new code would be much clearer than if it was four things looking entirely different but actually beeing more like two things where one has three variations.

We could even extract the `handleEnrollments` function now, with method documentation for that saying what it does and a comment beside that call in the main function saying that entirely different things are possible but not yet done.

**Comment by me97esn** [comment]
Yes, you do have a good point. Perhaps we should do this changes after all.
